pkg/cloud/amazon/s3_storage.go

```diff
@@ -674,9 +712,10 @@ func (s *s3Storage) putUploader(ctx context.Context, basename string) (io.WriteC
 		input: &s3.PutObjectInput{
 			Bucket:               s.bucket,
 			Key:                  aws.String(path.Join(s.prefix, basename)),
-			ServerSideEncryption: nilIfEmpty(s.conf.ServerEncMode),
+			ServerSideEncryption: types.ServerSideEncryption(s.conf.ServerEncMode),
 			SSEKMSKeyId:          nilIfEmpty(s.conf.ServerKMSID),
-			StorageClass:         nilIfEmpty(s.conf.StorageClass),
+			StorageClass:         types.StorageClass(s.conf.StorageClass),
+			ChecksumAlgorithm:    types.ChecksumAlgorithmSha256,
```

Comment:
We may want to follow up and figure out if this is resulting in _another_ sha256 to be taken or if the same one used for singing is used.